koap-core/src/commonMain/kotlin/Block.kt

```diff
@@ -0,0 +1,33 @@
+package com.juul.koap
+
+import kotlin.reflect.KClass
+
+internal val Message.Option.Block.intValue: Int
+    get() {
+        val moreBit = if (more) 1 else 0
+        return (number shl 4) or (moreBit shl 3) or size.intValue
+    }
+
+internal inline fun <reified T : Message.Option.Block> blockOf(value: Long): T =
+    blockOf(T::class, value)
+
+private fun <T : Message.Option.Block> blockOf(type: KClass<T>, value: Long): T {
+    val number = value shr 4
+    require(number in BLOCK_NUMBER_RANGE) {
+        "Block number $number (from option value of $value) is outside allowable range of $BLOCK_NUMBER_RANGE"
+    }
+    val more = (value and 0x8L) != 0L
+    val size = (value and 0b111).toInt().let {
+        Message.Option.Block.Size.entries
+            .first { entry -> entry.intValue == it }
```

Comment:
could rely/require the Sizes to be defined with ordinal == szx value, and use Block.Size.entries[value and 0b111] for lookup?